unit_scaling/_modules.py

```diff
@@ -313,34 +573,34 @@ def __init__(
         vocab_size: int,
         layers: int,
         heads: int,
-        dropout_p: float,
+        dropout_p: float = 0.0,
         act_fn: nn.Module = GELU(),
-        mhsa_tau: float = 0.01,
-        mlp_tau: float = 0.5,
-        constraint: Optional[str] = "gmean",
+        residual_scaling: ResidualScalingFn = transformer_residual_scaling_rule(),
+        constraint: Optional[str] = "to_output_scale",
     ) -> None:
         super().__init__()
+        # Layers are applied in order
         self.embedding = Embedding(vocab_size, hidden_size)
-        self.dropout_p = dropout_p
-        self.initial_layer_norm = LayerNorm(hidden_size)
-        self.transformer_layers = nn.Sequential(
-            *(
-                TransformerLayer(
-                    hidden_size, heads, dropout_p, act_fn, mhsa_tau, mlp_tau, constraint
-                )
-                for _ in range(layers)
-            )
+        self.embedding_dropout = Dropout(dropout_p)
```

Comment:
I don't think we have dropout or norm after embedding usually? (based on https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama/model.py#L417-L436)

Reply:
Agree, thanks - a shared-embedding hangover I think, I've deleted it.